src/test/transaction_tests.cpp

```diff
@@ -367,6 +366,27 @@ BOOST_AUTO_TEST_CASE(tx_invalid)
     }
 }
 
+BOOST_AUTO_TEST_CASE(tx_no_inputs)
+{
+    TxValidationState state;
+    bool isValid = CheckTransaction(CTransaction(CMutableTransaction()), state);
+
+    BOOST_CHECK_MESSAGE(!isValid, "Transaction with no inputs should be invalid.");
+    BOOST_CHECK(state.GetRejectReason() == "bad-txns-vin-empty");
+}
+
+BOOST_AUTO_TEST_CASE(tx_oversized)
+{
+    CMutableTransaction tx;
+    tx.vin.resize(1);
+    auto largeOutput = CScript() << OP_RETURN << std::vector<unsigned char>(MAX_BLOCK_WEIGHT / WITNESS_SCALE_FACTOR, 0x00);
```

Comment:
To check see that the test would fail without "bad-txns-oversize", modified `tx_oversized` case temporarily with the following:

```c++
auto largeOutput = CScript() << OP_RETURN << std::vector<unsigned char>(60, 0x00);
```

Unit test failed (as expected).

Reply:
thanks for checking